lib/models/device.coffee

```diff
@@ -642,6 +642,71 @@ getDeviceModel = (deps, opts) ->
 		.asCallback(callback)
 
 	###*
+	# @summary Set a custom location for a device
+	# @name setCustomLocation
+	# @public
+	# @function
+	# @memberof resin.models.device
+	#
+	# @param {String|Number} uuidOrId - device uuid (string) or id (number)
+	# @param {Object} location - the location ({ latitude: 123, longitude: 456 })
+	#
+	# @returns {Promise}
+	#
+	# @example
+	# resin.models.device.setCustomLocation('7cf02a6', { latitude: 123, longitude: 456 });
+	#
+	# @example
+	# resin.models.device.setCustomLocation(123, { latitude: 123, longitude: 456 });
+	#
+	# @example
+	# resin.models.device.setCustomLocation('7cf02a6', { latitude: 123, longitude: 456 }, function(error) {
+	# 	if (error) throw error;
+	# });
+	###
+	exports.setCustomLocation = (uuidOrId, location, callback) ->
+		exports.get(uuidOrId).then (device) ->
+			return pine.patch
+				resource: 'device'
+				body:
+					custom_latitude: String(location.latitude),
```

Comment:
I've mainly done it this way simply because that's what the existing UI code currently does and expects. Not sure if there's a reason we can't use nulls here, that sounds reasonable to me. @LucianBuzzo?